dogstatsd.py

```diff
@@ -131,6 +133,43 @@ def serialize_event(event):
     return json.dumps(event)
 
 
+def mapto_v6(addr):
+    """
+    Map an IPv4 address to an IPv6 one.
+    If the address is already an IPv6 one, just return it.
+    Return None if the IP address is not valid.
+    """
+    try:
+        inet_pton(socket.AF_INET, addr)
+        return '::ffff:{}'.format(addr)
+    except socket.error:
+        try:
+            inet_pton(socket.AF_INET6, addr)
+            return addr
+        except socket.error:
+            log.debug('%s is not a valid IP address.', addr)
+
+    return None
+
+
+def normalize_host(host):
+    """
+    Map the host to an IPv6 address, trying to resolve the name
+    """
+    try:
+        addr = socket.gethostbyname(host)
```

Comment:
Not that it matters much because you'll be converting the ipv4 to ipv6 anyways - but maybe we should favor the ipv6 record if it exists. I'm saying this because the DNS might have the same name resolve to an `A` (ipv4) and an `AAAA` (ipv6) record.... So `getaddrinfo` might be a better option for dual-stack resolution - otherwise if the hostname only has an ipv6 record this might fail.